### PR TITLE
ISPN-15199 Expose needed parameters for the console

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-15199

### CODE REVIEW

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 1]
```diff
@@ -763,6 +765,15 @@ private RestResponse getDetailResponse(RestRequest request, Cache<?, ?> cache, b
       fullDetail.rebalancingEnabled = rebalancingEnabled;
       fullDetail.keyStorage = cache.getAdvancedCache().getKeyDataConversion().getStorageMediaType();
       fullDetail.valueStorage = cache.getAdvancedCache().getValueDataConversion().getStorageMediaType();
+      fullDetail.mode = configuration.clustering().cacheModeString();
+      fullDetail.keyEncoding = configuration.encoding().keyDataType().mediaType();
+      fullDetail.valueEncoding = configuration.encoding().keyDataType().mediaType();
+
+      if (invocationHelper.getRestCacheManager().getAuthorizer().getPermissions(null, request.getSubject()).contains(AuthorizationPermission.ADMIN)) {
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:772]: I'd prefer if you used:
```java
authorizationManager.doIf(request.getSubject(), AuthorizationPermission.ADMIN, () -> { ... });
```

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheResourceV2Test.java [hunk 2]
```diff
@@ -695,8 +695,15 @@ public void testCacheFullDetail() {
...
-      assertThat(document.at("value_storage").asString()).isEqualTo("application/unknown");
+      assertThat(document.at("key_storage").asString()).isEqualTo("application/x-protostream");
+      assertThat(document.at("value_storage").asString()).isEqualTo("application/x-protostream");
+      assertThat(document.at("key_encoding").asString()).isEqualTo("application/x-protostream");
+      assertThat(document.at("value_encoding").asString()).isEqualTo("application/x-protostream");
+      assertThat(document.at("mode").asString()).isEqualTo("DIST_SYNC");
+      assertThat(document.at("storage_type").asString()).isEqualTo("HEAP");
+      assertThat(document.at("max_size").asString()).isEmpty();
+      assertThat(document.at("max_size_bytes").asLong()).isEqualTo(-1);
+
```
Comment (ryanemerson) [inline server/rest/src/test/java/org/infinispan/rest/resources/CacheResourceV2Test.java:706]: Should we also adds assertion for when `security == true` to ensure that `storageType`, `maxSize` and `maxSizeBytes` are not set?

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheResourceV2Test.java:706]: done @ryanemerson

**Comment by ryanemerson** [comment]
@karesti Please rebase on the latest `main`, we need the changes to the Jenkinsfile in order for CI to report failures correctly.

**Comment by karesti** [comment]
@ryanemerson done

**Comment by karesti** [comment]
@ryanemerson build seems ok, can we please merge https://ci.infinispan.org/job/Infinispan/job/PR-11342/7/

**Comment by ryanemerson** [comment]
Thanks @karesti

**Pull request merged**